docs/sql-ref-literals.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: global
+title: Literals
+displayTitle: Literals
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+A literal (aka constant) represents a fixed data value. Spark SQL supports the following literals:
+
+ * String Literals
+ * Boolean Literals
+ * Numeric Literals
+ * Datetime Literals
+ * Interval Literals
+
+### String Literals
+
+A string literal is used to specify a character string value.
+<dl>
+  <dt><code><em>Format:</em></code></dt>
+  <dd>
+    <code>'c [ ... ]'</code><br>
+    c: one character of user's character set.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>Examples:</em></code></dt>
+  <dd>
+    <code>'Hello, World!',   'Spark SQL',   'dbname.schema'</code>
+  </dd>
+</dl>
+
+### Boolean Literals
+
+A boolean literal is used to specify a boolean value.
+<dl>
+  <dt><code><em>Format:</em></code></dt>
+  <dd>
+    <code>TRUE | FALSE </code>
+  </dd>
+</dl>
+
+### Numeric Literals
+
+A numeric literal is used to specify a fixed or floating-point number.
+
+#### Integer Literals
+
+<dl>
+  <dt><code><em>Format:</em></code></dt>
+  <dd>
+    <code>[ + | - ] digit [ ... ] [ l | L ]</code><br>
+    digit: one of 0, 1, 2, 3, 4, 5, 6, 7, 8 or 9 <br>
+    l or L: indicates <code>LongType</code>.
```

Comment:
can we document all the postfixes? And also mention that no postfix means integer type.